pyro/infer/abstract_infer.py

```diff
@@ -317,8 +333,12 @@ def _adjust_to_data(self, trace, data_trace):
                     # Select random sub-indices to replay values under conditionally independent stacks.
                     # Otherwise, we assume there is an dependence of indexes between training data
                     # and prediction data.
-                    subidxs = Categorical(logits=site["value"].new_ones(site["value"].size(cis.dim))).sample([cis.size])
-                    site["value"] = site["value"].index_select(cis.dim, subidxs)
+                    bdim = cis.dim - site["fn"].event_dim
+                    cat = Categorical(logits=torch.ones(site["value"].size(bdim),
+                                                        device=site["value"].device,
+                                                        dtype=torch.float))
+                    subidxs = cat.sample((cis.size,))
+                    site["value"] = site["value"].index_select(bdim, subidxs)
```

Comment:
I think we discussed this the last time, and I am forgetting now, but can we not simply use:

```python
subidxs = torch.randint(0, site['value'].size(bdim), cis.size)
```

Reply:
Thanks! I think it seems to work. I probably just thought in terms of distributions when I made the first PR 😄